egs/wsj/s5/steps/nnet3/align_lats.sh

```diff
@@ -92,12 +92,27 @@ if [ -f $srcdir/frame_subsampling_factor ]; then
   frame_subsampling_factor=$(cat $srcdir/frame_subsampling_factor)
   frame_subsampling_opt="--frame-subsampling-factor=$frame_subsampling_factor"
   cp $srcdir/frame_subsampling_factor $dir
-  if [ "$frame_subsampling_factor" -gt 1 ] && \
-     [ "$scale_opts" == "--transition-scale=1.0 --self-loop-scale=0.1" ]; then
-    echo "$0: frame-subsampling-factor is not 1 (so likely a chain system),"
-    echo "...  but the scale opts are the defaults.  You probably want"
-    echo "--scale-opts '--transition-scale=1.0 --self-loop-scale=1.0'"
-    sleep 1
+  if [[ $frame_subsampling_factor -gt 1 ]]; then
+    # Likely a chain system, check agrument sanity.
+    epilogue() { :; }
```

Comment:
Sorry this is just a little bit too much for me; I'd prefer the code be simple and self-explanatory.
Just printing the error message and exiting with status 1 is fine; people can always set slightly different values if they really want to use those (weird) values.

Reply:
This is what it did before w.r.t. the scale_opts, printed a warning and paused, did not exit. I just added the second check. Am I missing the idea?

Reply:
You can simplify it and have it just exit.  All this epilogue() / prologue() stuff is too complex-- just have it print the message and exit.

Reply:
Or do you mean the code style? Local helper functions?

Reply:
Exiting makes sense only we sure that this is in fact a chain model--are we?

I can redo the same logic without functions, if you think they are unreadable. To me, they are just shell variables, only functions, but I'm not going to insist. :)

Maybe it would be better then make a separate nnet3/chain/aligh_lats.sh, with the correct defaults? Or nor even defaults, just hardcoded required values. What do you think?